### PR TITLE
Sidebar: v-2 copy version

### PR DESCRIPTION
**Pull request by tishoyanchev**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

A copy version of 502-sidebar-v-2 branch

- added 3 layers of menu-item nesting
- added border-indicator
- added the number indicator and arrow
- Updated storybook to showcase the new changes

<details>
  <summary>📦 Published PR as canary version: <code>20.38.0--canary.791.2e995182215d32a11a88ecc0768916413321b709.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @infineon/infineon-design-system-stencil@20.38.0--canary.791.2e995182215d32a11a88ecc0768916413321b709.0
  # or 
  yarn add @infineon/infineon-design-system-stencil@20.38.0--canary.791.2e995182215d32a11a88ecc0768916413321b709.0
  ```
</details>

### CODE REVIEW

**Review by verena-ifx — COMMENTED** [review]
we definitely need to add documentation (probably needs to be the custom .mdx story, informing the user how to configure the sidebar items (listing the props of each sidebar-item with a description)
besides that it looks great!

**Comment by tishoyanchev** [comment]
> we definitely need to add documentation (probably needs to be the custom .mdx story, informing the user how to configure the sidebar items (listing the props of each sidebar-item with a description) besides that it looks great!

I agree, an additional explanation on how the **active** sidebar-items work along with the **number-indicator** would be a good idea.

**Review by verena-ifx — COMMENTED** [review]
also, can you please add e2e tests?

**Comment by verena-ifx** [comment]
> > also, can you please add e2e tests?
> 
> There is a e2e test.

okay i will check it out then, didnt see it in the changed files, but I guess it was already there then! My bad!

**Review by verena-ifx — COMMENTED** [review]
- active state doesnt get removed, when another item is selected;
- storybook is missing the action for when an item is selected

File: packages/components/src/components/sidebar/sidebar-item.tsx [hunk 1]
```diff
@@ -22,20 +46,199 @@ export class SidebarItem {
     }
   }
 
+  handleEventEmission() { 
+    this.ifxSidebarItem.emit({value: this.value, component: this.el})
```
Comment (verena-ifx) [inline packages/components/src/components/sidebar/sidebar-item.tsx:50]: this.value is never set and always ""

Comment (tishoyanchev) [reply packages/components/src/components/sidebar/sidebar-item.tsx:50]: It is emitted when you click on an item. 
Check function **toggleSubmenu()**, which is invoked on click. 
 `onClick={() => this.toggleSubmenu()}`

**Review by benedikt-ifx — APPROVED** [review]
Looks great

**Pull request merged**